src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/actions/BaseFlowRuleRemovalAction.java

```diff
@@ -235,4 +238,16 @@ protected PathContext buildPathContextForRemovalIngressOnly(SwitchId switchId) {
                 .server42MacAddress(switchProperties.getServer42MacAddress())
                 .build();
     }
+
+    protected Flow getOriginalFlowWithPaths(FlowPathSwappingFsm stateMachine, RequestedFlow originalFlow) {
```

Comment:
By design and implementation, BaseFlowRuleRemovalAction is supposed to be used with any FSM which extends FlowProcessingFsm. But this method requires FlowPathSwappingFsm to be provided. Can we align with initial design?

Reply:
Reworked.